integration/ide/bsp-server/src/BspServerTests.scala

```diff
@@ -0,0 +1,158 @@
+package mill.integration
+
+import ch.epfl.scala.{bsp4j => b}
+import mill.api.BuildInfo
+import mill.bsp.Constants
+import mill.testkit.BspServerTestUtil._
+import mill.testkit.UtestIntegrationTestSuite
+import utest._
+
+import scala.jdk.CollectionConverters._
+
+object BspServerTests extends UtestIntegrationTestSuite {
+  def snapshotsPath: os.Path =
+    super.workspaceSourcePath / "snapshots"
+  override protected def workspaceSourcePath: os.Path =
+    super.workspaceSourcePath / "project"
+
+  def tests: Tests = Tests {
+    test("BSP install") - integrationTest { tester =>
```

Comment:
We should rename this to `snapshotTests` or something

Reply:
Or `requests`? This tests that the main BSP requests work fine. `project structure` might work too, this tests the "descriptive" BSP requests, that expose the project structure somehow.

Reply:
`requestSnapshotTests`? I think being a bit verbose here is fine, since the suite is definitely doing something unusual

Reply:
That would be `requestSnapshots` in the former case, right? No need of the `Test` suffix

Reply:
Renamed them the test to `requestSnapshots`